ui/lib/pki/addon/components/page/pki-issuer-rotate-root.ts

```diff
@@ -0,0 +1,115 @@
+import Component from '@glimmer/component';
+import { inject as service } from '@ember/service';
+import { tracked } from '@glimmer/tracking';
+import { action } from '@ember/object';
+import { parseCertificate } from 'vault/utils/parse-pki-cert';
+import camelizeKeys from 'vault/utils/camelize-object-keys';
+import { waitFor } from '@ember/test-waiters';
+import { task } from 'ember-concurrency';
+import errorMessage from 'vault/utils/error-message';
+// TYPES
+import Store from '@ember-data/store';
+import Router from '@ember/routing/router';
+import FlashMessageService from 'vault/services/flash-messages';
+import SecretMountPath from 'vault/services/secret-mount-path';
+import PkiIssuerModel from 'vault/models/pki/issuer';
+import { Breadcrumb } from 'vault/vault/app-types';
+import { parsedParameters } from 'vault/utils/parse-pki-cert-oids';
+
+interface Args {
+  oldRoot: PkiIssuerModel;
+  breadcrumbs: Breadcrumb;
+}
+
+export default class PagePkiIssuerRotateRootComponent extends Component<Args> {
+  @service declare readonly store: Store;
+  @service declare readonly router: Router;
+  @service declare readonly flashMessages: FlashMessageService;
+  @service declare readonly secretMountPath: SecretMountPath;
+
+  @tracked rotateForm = 'use-old-settings';
+  @tracked showOldSettings = false;
+  @tracked newRootModel;
+  @tracked alertBanner = '';
+  @tracked invalidFormAlert = '';
+
+  constructor(owner: unknown, args: Args) {
+    super(owner, args);
+    const certData = parseCertificate(this.args.oldRoot.certificate);
+    if (certData.parsing_errors && certData.parsing_errors.length > 0) {
+      const errorMessage = certData.parsing_errors.map((e: Error) => e.message).join(', ');
+      this.alertBanner = errorMessage;
+    }
+    this.newRootModel = this.store.createRecord('pki/action', {
```

Comment:
I had trouble getting the parent model passed down to the child component. 

But I'd like to refactor so that we can use the `withConfirmLeave()` decorator